caddyhttp/rewrite/setup_test.go

```diff
@@ -175,13 +175,23 @@ func TestRewriteParse(t *testing.T) {
 			}
 
 			if actualRule.Regexp != nil {
-				if actualRule.String() != expectedRule.String() {
+				if actualRule.Regexp.String() != expectedRule.Regexp.String() {
 					t.Errorf("Test %d, rule %d: Expected Pattern=%s, got %s",
-						i, j, expectedRule.String(), actualRule.String())
+						i, j, actualRule.Regexp.String(), expectedRule.Regexp.String())
 				}
 			}
-
 		}
+
+		testRewriteParseString(t, actual, i)
 	}
+}
 
+func testRewriteParseString(t *testing.T, rules []httpserver.HandlerConfig, testIndex int) {
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("Test %d: Failed to string encode: %#v", testIndex, r)
+		}
+	}()
```

Comment:
Does this test actually work? I was under the impression that `fmt` kind of recovered from panics by encoding the panic into a string.